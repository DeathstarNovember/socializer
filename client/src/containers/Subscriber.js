import { useEffect } from "react";

const Subscriber = ({ subscribeToNew, children }) => {
  useEffect(() => {
    subscribeToNew();
  }, [subscribeToNew]);

  return children;
};

export default Subscriber;
