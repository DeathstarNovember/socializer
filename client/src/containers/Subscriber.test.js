import React from "react";
import { render } from "@testing-library/react";
import Subscriber from "./Subscriber";

describe("Subscriber", () => {
  it("invokes subscription and renders children", () => {
    const subscribeToNew = jest.fn();
    const { container } = render(
      <Subscriber subscribeToNew={subscribeToNew}>
        <p>Hello</p>
      </Subscriber>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(subscribeToNew).toBeCalled();
  });
});
