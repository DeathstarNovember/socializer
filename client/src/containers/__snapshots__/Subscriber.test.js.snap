// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscriber invokes subscription and renders children 1`] = `
<p>
  Hello
</p>
`;
